test/dynamo/test_hooks.py

```diff
@@ -316,17 +320,55 @@ def forward(self, x):
         aot_out[0].backward(torch.ones(4))
 
         x2 = torch.ones(4, requires_grad=True)
-        dynamo_out = torch._dynamo.optimize("aot_eager")(mod)(x2)
-        dynamo_out[0].backward(torch.ones(4))
+        dynamo_out = torch._dynamo.optimize("aot_eager", nopython=True)(mod)(x2)
+        with compiled_autograd.enable(compiler_fn):
+            dynamo_out[0].backward(torch.ones(4))
 
         self.assertEqual(dynamo_out, aot_out)
         self.assertEqual(dynamo_out, eager_out)
 
         self.assertEqual(x0.grad, x1.grad)
         self.assertEqual(x0.grad, x2.grad)
 
+    def test_input_hooks_same(self):
```

Comment:
it seems... weird that you're adding a test for the input hook case here? Isn't this PR about intermediate hooks

Reply:
Yeah, but this is a new case I thought of. I dont mind giving it its own PR.